static/app/utils/useProjects.tsx

```diff
@@ -0,0 +1,416 @@
+import {useEffect, useRef, useState} from 'react';
+import memoize from 'lodash/memoize';
+import partition from 'lodash/partition';
+import uniqBy from 'lodash/uniqBy';
+
+import ProjectActions from 'app/actions/projectActions';
+import {Client} from 'app/api';
+import ProjectsStore from 'app/stores/projectsStore';
+import {AvatarProject, Project} from 'app/types';
+import {defined} from 'app/utils';
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+import RequestError from 'app/utils/requestError/requestError';
+import useApi from 'app/utils/useApi';
+
+type ProjectPlaceholder = AvatarProject;
+
+type State = {
+  /**
+   * Projects from API
+   */
+  fetchedProjects: Project[] | ProjectPlaceholder[];
+  /**
+   * Projects fetched from store
+   */
+  projectsFromStore: Project[];
+  /**
+   * Reflects whether or not the initial fetch for the requested projects
+   * was fulfilled
+   */
+  initiallyLoaded: boolean;
+  /**
+   * This is state for when fetching data from API
+   */
+  fetching: boolean;
+  /**
+   * This is set when we fail to find some slugs from both store and API
+   */
+  isIncomplete: null | boolean;
+  /**
+   * Project results (from API) are paginated and there are more projects
+   * that are not in the initial queryset
+   */
+  hasMore: null | boolean;
+  prevSearch: null | string;
+  nextCursor?: null | string;
+  /**
+   * The error that occurred if fetching failed
+   */
+  fetchError: null | RequestError;
+};
+
+export type Result = {
+  /**
+   * We want to make sure that at the minimum, we return a list of objects with
+   * only `slug` while we load actual project data
+   */
+  projects: Project[] | ProjectPlaceholder[];
+  /**
+   * Calls API and searches for project
+   */
+  onSearch: (searchTerm: string, {append: boolean}) => void;
+} & Pick<
+  State,
+  'isIncomplete' | 'fetching' | 'hasMore' | 'initiallyLoaded' | 'fetchError'
+>;
+
+type Options = {
+  /**
+   * Organization slug
+   */
+  orgId: string;
+  /**
+   * List of slugs to look for summaries for, this can be from `props.projects`,
```

Comment:
Is there still a `props.projects`?

Reply:
This was previously used to inject projects from `withProjects`. I've removed the prop since it wasn't being used anywhere to override the projects list and have inlined the store listener into the hook itself.